packages/spartan/extra/gen_resolvconf.py

```diff
@@ -69,7 +69,20 @@ def check_server(addr):
 
 # If Spartan is not up, fall back, and insert the upstreams
 else:
-    fallback_servers = os.environ['RESOLVERS'].split(',')
+    fallback_servers = []
+
+    # Resolvconf does not support custom ports, skip if not default
+    for ns in os.environ['RESOLVERS'].split(','):
+        ip, separator, port = ns.rpartition(':')
+        if not separator:
+            fallback_servers.append(ns)
+            continue
+        if port == "53":
+            fallback_servers.append(ip)
```

Comment:
This doesn't seem to match the docs: https://docs.python.org/3.4/library/stdtypes.html#str.rpartition